### ISSUE TITLE
Support for user generated content applications

### ISSUE THREAD

**Issue by nedrichards** [hughsie/oars#1]
Whilst looking to add [lbry to flathub](https://github.com/flathub/flathub/pull/175), there were questions about how to make OARS better able to cover user generated content applications:

> Should OARS tags be added to the appdata file and if so how? Content on LBRY is created by users so it can contain any of the content warnings found in OARS.

> What isn't subject to the content are these settings found in the app:

> "Share Diagnostic Data" is enabled by default
"Show NSFW content" is disabled by default
Maybe OARS needs a new tag like social-content with None, Regulated, Unregulated? This might be better discussed on the the OARS bugtracker (if they have one).

**Comment by hughsie** [comment]
I've added two more categories to `social-info` for the Diagnostic Data kind of thing, splitting it up into different classes depending if it's identifiable to the user or not.

If the user is able to "enable" NSFW content, then this should be included in the assessment even if it is turned off by default. I've included that as an example in the introduction now.

I've switched `social-chat=moderate` to include the moderated word to cope with social interactions that have a moderator, and left  `social-chat=intense` as unmoderated.

Ideas welcome about how to expand this for things like a twitter client.

**Comment by hughsie** [comment]
Take a look at https://hughsie.github.io/oars/ for a sneak peak.

**Comment by julianrichen** [comment]
For "Application that can connect to the Internet" I think it could use a few more tweaks.

**Money -> In-App Purchases**
> Users are encouraged to donate to the project

Should become "Users are encouraged to donate". At least in LBRY you don't donate to the "project" (LBRY.io). You use the LBRY currency to pay content creators. It's similar to supporting them via Patreon, YouTube , Twitch, etc...

> Ability to spend real money in-game, e.g. buying new content or new levels

Still has "in-game" as opposed to "in-app".

**Social -> Online Audio and Video Messaging**
> Uncontrolled audio or video chat between users

Maybe change this or create a new option tailored to non-chat? In LBRY users don't chat, they post their audio, videos, pictures, etc.. for anyone to anonymously see. Other apps like [gnome-twitch](https://github.com/vinszent/gnome-twitch) could benefit since the client shows streamers. They don't technically chat, they stream content for anyone to watch.

**Violence, Drugs, Language, Sex and Nudity**

LBRY could potently allow users to see violent content which features drugs, language, sex, and nudity. Allowing these options on the website when you select "Application that can connect to the Internet" might be a good idea?

Or maybe offer a more general "NSFW Content" option with none, moderate, and intense?
none = App contains no NSFW content
moderate = App may contain NSFW content
intense = App contains NSFW content

@hughsie Thanks again for looking into this so quickly.

**Comment by hughsie** [comment]
> Should become "Users are encouraged to donate

Fixed, thanks.

> Still has "in-game" as opposed to "in-app".

Fixed, thanks.

> Maybe change this or create a new option tailored to non-chat

I think this is covered in "social-audio" -- i.e. even LBRY is a "multimedia messaging system" -- and depending on if it's moderated we can choose a value.

> App contains no NSFW content

OARS is all about defining what *specifically* makes it NSFW, rather than trying to define an umbrella superset term. I'm totally up for expanding OARS to make it more useful but we do need to be specific.

**Comment by julianrichen** [comment]
> I think this is covered in "social-audio" -- i.e. even LBRY is a "multimedia messaging system" -- and depending on if it's moderated we can choose a value.

Yes, you are right. I read messaging but missed the multimedia part.

> OARS is all about defining what specifically makes it NSFW, rather than trying to define an umbrella superset term. I'm totally up for expanding OARS to make it more useful but we do need to be specific.

Yes you are right.

---

I've generated the xml, if I add the code to the appdata file now will LBRY only be visible on GNOME Software >= 3.27 or will the app still be visible on older versions and ignore tags it doesn't know?

Simply, will using the new xml break anything?

**Comment by hughsie** [comment]
It'll break nothing, but it the new categories won't work in gnome-software until the translations have been added.
